TempleEight/docs/arch/system.md

```diff
@@ -3,3 +3,189 @@ id: system
+    baz: int;
+}
+```
+
+This has the effect of generating inter-service communication logic in the service-level application code.
+On the project level, nothing changes, except the services use the hostname resolution of the orchestration platform to make HTTP requests to each other.
+
+<img alt="System Architecture Kube" src={useBaseUrl('img/system-architecture-foreign-key.png')} />
+
+### Structs
+
+Adding a struct to a service in your Templefile represents two data items that are strongly coupled (see the [Struct](../guide/structs) guide for details).
+This is implemented in Temple generated projects by extending the parent's application executable to include the additional endpoints required to process struct requests.
+The struct's data is stored in a separate table of the service's database. 
+This means that when adding structs to your Templefile, the overall system-level architecture doesn't change.
+
+### Auth
+
+As per the [Authentication](../guide/authentication) guide, 
+authentication can be added to your system by augmenting your Templefile to include `#auth` metadata on services, and a `#authMethod` annotation on the project block.
+For example:
+
+```templefile {5,10}
+ExampleProject: project {
+  #language(go);
+  #database(postgres);
+  #provider(dockerCompose);
+  #authMethod(email);
+}
+
+ExampleService: service {
+  foo: string;
+  bar: int;
+  #auth;
+}
+```
+
+These changes result in an additional `Auth` service being generated automatically that stores and verifies user credentials, and issues authentication tokens to users for use with other services.
+This is implemented by configuring Kong to check that any incoming requests have a valid [JWT](https://jwt.io/), except those destined for the auth service. 
+Any requests that are not correctly authenticated do not pass Kong, so the internal microservices can be sure that all traffic has been correctly authenticated.
+
+Visually, the changes to our architecture look like:
+
+<img alt="System Architecture Kube" src={useBaseUrl('img/system-architecture-auth.png')} />
+
+### Metrics
+
+Temple also includes the ability to generate metrics monitoring suites, using [Prometheus](https://prometheus.io/) and [Grafana](https://grafana.com/).
+For full details of their use and implementations, check out the [Metrics](../guide/metrics) guide.
+
+Let's add metrics to our example and see how the output changes.
+
+```templefile {5}
+ExampleProject: project {
+  #language(go);
+  #database(postgres);
+  #provider(dockerCompose);
+  #metrics(prometheus);
+}
+
+ExampleService: service {
+  foo: string;
+  bar: int;
+}
+```
+
+When a project has metrics, two more infrastructure components are needed, a Prometheus instance and a Grafana instance. 
+Prometheus is a tool for aggregating metrics from all your services into one centralised place. 
+Each service contains logic to aggregate metrics about it's performance, which are pulled into Prometheus ready for querying.
```

Comment:
I'm bad at English, this correct use of it's? @xsanda

Reply:
I really need to educate myself on that one

Reply:
It’s = it is
Its = it belongs to

Reply:
Oh cool, for some reason I always thought there was an edge case  where `it's` is possessive. I can now live my life without avoiding it  😂 a new dawn